lib/index.js

```diff
@@ -74,15 +121,40 @@ function compileBody(str, options){
         return filename;
       },
       read: function (filename) {
+        var str;
+        var plugins = getPlugin('loadRead');
+        for (var i = 0; i < plugins.length; i++) {
+          str = plugins[i](filename, source, options, dependencies);
+          if (str) {
```

Comment:
The empty string should be considered file content here.  Lets do `str !== undefined` as our test.